daemon/daemon_main.go

```diff
@@ -1585,11 +1586,17 @@ func (d *Daemon) launchHubble() {
 		logger.WithError(err).Warn("Failed to initialize Hubble")
 		return
 	}
-	s := hubbleServer.NewLocalServer(payloadParser, option.Config.HubbleFlowBufferSize, option.Config.HubbleEventQueueSize, logger)
+	s, err := hubbleServer.NewLocalServer(payloadParser, logger,
+		serveroption.WithMaxFlows(option.Config.HubbleFlowBufferSize),
+		serveroption.WithMonitorBuffer(option.Config.HubbleEventQueueSize))
+	if err != nil {
+		logger.WithError(err).Warn("Failed to initialize Hubble")
+		return
+	}
 	go s.Start()
 	d.monitorAgent.GetMonitor().RegisterNewListener(context.TODO(), hubble.NewHubbleListener(s))
 	logger.WithField("addresses", addresses).Info("Starting Hubble server")
-	if err := hubbleServe.Serve(logger, addresses, s); err != nil {
+	if err := hubbleServe.Serve(context.Background(), logger, addresses, s); err != nil {
```

Comment:
I think that the daemon context  (`d.ctx`) should be passed to `hubbleServe.Serve()` instead of `context.Background()`. This will allow a proper tear-down of hubble server when cilium is shutting down.

Reply:
done